tests/checks/webconfig-python-compat.fish

```diff
@@ -0,0 +1,27 @@
+#RUN: %fish %s
+#REQUIRES: command -v uvx
+
+set -l webconfig (status dirname)/../../share/tools/web_config/webconfig.py
+set -l min_version 3.5
+
+# use vermin to detect minimum Python version violations
+# features enabled:
+#   - union-types: catch `str | None` syntax (3.10+)
+#   - fstring-self-doc: catch f'{var=}' syntax (3.8+)
+set -l output (uvx vermin \
+    --no-tips \
+    --no-parse-comments \
+    --violations \
+    --feature union-types \
+    --feature fstring-self-doc \
+    --target=$min_version- \
+    $webconfig \
```

Comment:
Awesome work, LGTM!

If you want, you could also do
share/tools/create_manpage_completions.py
which is the only other user-facing python script.